client/files.go

```diff
@@ -110,3 +115,56 @@ func WriteUserSignedKey(signedKey string, key string) (err error) {
 	_, err = signedKeyFile.WriteString(signedKey)
 	return err
 }
+
+// FindUserPubKeys checks every pubkey in `keys` list and returns only existsing keys (or error if
+// all pubkeys don't exist)
+func FindUserPubKeys(keys []string) ([]string, error) {
+	var found = []string{}
+	for _, key := range keys {
+		pubKeyPath, err := UserPubKeyExists(key)
+		if err != nil {
+			return nil, err
+		}
+		if pubKeyPath != "" {
+			found = append(found, pubKeyPath)
+		}
+	}
+
+	if len(found) == 0 {
+		var errStr = fmt.Sprintf(`user SSH keys at %s doesn't exist.
+
+Please generate at least one with command like this :
+
+`, strings.Join(keys, ", "))
+
+		// keys list is not explicitly set, suggest generating ed25519 as default
+		if reflect.DeepEqual(keys, DefaultSSHKeys) {
+			errStr += "\tssh-keygen -f ~/.ssh/id_ed25519 -t ed25519\n"
+		} else {
+			for _, key := range keys {
+				suggestedSSHKey := strings.Replace(key, ".pub", "", 1)
+				errStr += fmt.Sprintf("\tssh-keygen -f %s -t %s\n", suggestedSSHKey, chooseSSHKeyType(suggestedSSHKey))
+			}
+		}
+
+		return nil, fmt.Errorf(errStr)
+	}
+	return found, nil
+}
+
+func chooseSSHKeyType(key string) string {
+	switch {
+	case strings.Contains(key, "dsa"):
+		return "dsa"
```

Comment:
I think it could be good to add a message saying it's deprecated and the use of ed25519 is preferred ?

Reply:
Added deprecation warning

```
ssh-keygen -f /home/podkorytov_pm/Work/signmykey/ssh/2.dsa -t dsa [deprecated, consider using ed25519 instead]
```